### ISSUE TITLE
How to combine with express-validation?

### ISSUE THREAD

**Issue by georgiosd** [vkarpov15/awaitjs-express#6]
Sadly, I get an error when I use it in combination with the `express-validation` package from https://github.com/AndrewKeig/express-validation

Error:

```
TypeError: Cannot read property 'then' of undefined
    at wrappedMiddleware (C:\Users\Georgios\Documents\GitHub\node_modules\@awaitjs\express\index.js:100:23)
    at Layer.handle [as handle_request] (C:\Users\Georgios\Documents\GitHub\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\Georgios\Documents\GitHub\node_modules\express\lib\router\route.js:137:13)
```

**Comment by georgiosd** [comment]
I had to modify the source to only `wrap` the last arguments instead of all of them. Not sure if it's something you wanna integrate.

**Comment by 0xCAFEADD1C7** [comment]
This should be fixed by the PR above. 
Unfortunatly, It might still not work for other libraries which call `next` asynchronously. For example, the code below won't work as expected : 
```javascript
// middleware
function myMiddleware(req, res, next) {
  setTimeout(() => {
    req.ran = true;
    next();
  }, 100);
})

// route definition
app.getAsync('/', myMiddleware, function(req, res, next) { 
  console.log(req.ran); // this will print "undefined'
  res.send({});
})
```
I can't see any obvious way of fixing this behaviour...

**Comment by georgiosd** [comment]
@0xCAFEADD1C7 generally it's not possible to detect async functions reliably, or so StackOverflow tells me. HOWEVER, it seems that if you print out an async function in node, it gets printed as `AsyncFunction`, which seems detectable to me. 

Might be something to look into.

**Comment by vkarpov15** [comment]
@georgiosd I merged #7, which should fix this.

Also, for the purposes of this library we don't need to detect an async function, we just need to detect whether the function returns a promise. All async functions return promises so that's really all we care about. I like @0xCAFEADD1C7 's solution to use async/await though - `await` on a non-promise is a no-op.
